conda/install.py

```diff
@@ -839,12 +838,21 @@ def extract(dist):
     fname = rec['files'][0]
     assert url and fname
     pkgs_dir = dirname(fname)
+
     path = fname[:-8]
     with Locked(path):
         temp_path = path + '.tmp'
         rm_rf(temp_path)
+
+        # check whether has enough space to extract a file
+        from .fetch import check_size
         with tarfile.open(fname) as t:
+            size = 0
+            for m in t.getmembers():
+                size += m.size
+            check_size(pkgs_dir, size)
```

Comment:
I think this is the wrong place to do this.  I think we should sum the sizes of all the packages we're about to download and extract, and then do just one check.